lib/matplotlib/backend_bases.py

```diff
@@ -2440,6 +2477,19 @@ def stop_event_loop_default(self):
         """
         self._looping = False
 
+    def destroy(self):
+        pass
+
+    @property
+    def manager(self):
+        if self._manager is not None:
+            return self._manager()
+
+    @manager.setter
+    def manager(self, manager):
+        if manager is not None:
+            self._manager = weakref.ref(manager)
```

Comment:
To get rid of the circular reference, ``FigureManager`` holds a reference to ``Canvas`` and ``Canvas`` holds a reference to ``FigureManager``.  Trying to help GC do its job, it still doesn't lots more references to manager and I have no idea where they come from, but one step closer... circular references never good...